src/Storages/StorageReplicatedMergeTree.h

```diff
@@ -736,6 +763,10 @@ class StorageReplicatedMergeTree final : public shared_ptr_helper<StorageReplica
         std::unique_ptr<MergeTreeSettings> settings_,
         bool has_force_restore_data_flag,
         bool allow_renaming_);
+
+    /// Global ID, synced via ZooKeeper between replicas
+    /// mutable because can getted from ZooKeeper when required
+    mutable UUID table_global_id;
```

Comment:
Why we cannot create/get it in table constructor?

Reply:
We can. Fixed.